### PR TITLE
TECH Add Duration() function

### PR DESCRIPTION
**Pull request by jean-k-cp**
Will need this more modular approach than the Timer based on Now()

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -82,6 +82,21 @@ type Timing struct {
 	sender *Sender
 }
 
+// Duration can emit a timing metric from a duration.
+func Duration(duration time.Duration, bucket string) { sender.Duration(duration, bucket) }
```
Comment (hubertmine) [inline metrics.go]: ojo ! you inverted the signature we usually use

File: metrics_test.go [hunk 2]
```diff
@@ -448,3 +433,19 @@ func TestTiming_Success(t *testing.T) {
 	})
 	assert.Equal(t, "prefix1.test.timing:1000|ms", payload)
 }
+
+func setupTestUDPServer() (map[string]string, net.PacketConn, func() error) {
```
Comment (hubertmine) [inline metrics_test.go]: thx for the refacto !

File: metrics.go (hunk 1 above)
Comment (jean-k-cp) [inline metrics.go]: you're right ! fixed

File: metrics_test.go (hunk 2 above)
Comment (jean-k-cp) [inline metrics_test.go]: glad you liked it :blush:

**Pull request merged**